Formula/appunite-cache.rb

```diff
@@ -0,0 +1,21 @@
+class AppuniteCache < Formula
+  desc "AppUnite cache dependencies on S3 storage!"
+  homepage "https://git.appunite.com/szymon.mrozek/appunite-cache"
+  url "https://git.appunite.com/szymon.mrozek/appunite-cache.git",
+      :tag      => "1.0.6",
+      :revision => "e71d470acf0b2581903a8acd54a04dc7f8d4cdba",
+      :shallow  => false
+  head "https://git.appunite.com/szymon.mrozek/appunite-cache.git", :shallow => false
+
+  bottle do
+    cellar :any
+    sha256 "b5dc564ececccbb4102a2dc97ba76a717ecf900e7a867939aa0a64641d07165a" => :mojave
+  end
+
+  depends_on :xcode => ["10.0", :build]
+  depends_on "libressl"
+
+  def install
+    system "make", "prefix_install", "PREFIX=#{prefix}"
+  end
+end
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@SMillerDev Yeah my app is one of those which require credentials file to be configured. So as far as I understand I should create a test method but it should return `0` code anyway even for bad credentials?

Reply:
The homebrew formula should have a test block but it's well allowed to run commands with a non-zero exitcode

Reply:
done ✅

Reply:
@SMillerDev

Reply:
Can you make the test also verify something in the output? Something like this (I might be wrong on the exact syntax)
```suggestion
    assert_match "something", shell_output(bin/"appunite-cache verify-credentials")
  end
end
```

Reply:
It has two possible results so I can't always guarantee that it `asset_match` something ... Is it needed? I do not see it in `Carthage` for example: https://github.com/Homebrew/homebrew-core/blob/master/Formula/carthage.rb